packages/stateful/actions/components/TransferCw721.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  ArrowRightAltRounded,
+  SubdirectoryArrowRightRounded,
+} from '@mui/icons-material'
+import { useEffect, useMemo, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+
+import {
+  AddressInput,
+  ImageEmoji,
+  InputErrorMessage,
+  NftCard,
+  NftSelectionModal,
+} from '@dao-dao/stateless'
+import {
+  ActionComponent,
+  LoadingDataWithError,
+  NftCardInfo,
+} from '@dao-dao/types'
+import { validateAddress, validateRequired } from '@dao-dao/utils'
+
+import { ActionCard } from './ActionCard'
+
+export interface TransferCw721Options {
+  // The set of NFTs that may be transfered as part of this action.
+  options: LoadingDataWithError<NftCardInfo[]>
+  // Information about the NFT currently selected.
+  nftInfo: NftCardInfo | undefined
+}
+
+export const TransferCw721Component: ActionComponent<TransferCw721Options> = ({
+  fieldNamePrefix,
+  onRemove,
+  isCreating,
+  errors,
+  options: { options, nftInfo },
+}) => {
+  const { t } = useTranslation()
+  const { watch, setValue, setError, register, clearErrors } = useFormContext()
+
+  const tokenId = watch(fieldNamePrefix + 'tokenId')
+  const collection = watch(fieldNamePrefix + 'collection')
+
+  const selected = useMemo(
+    () => `${collection}:${tokenId}`,
+    [tokenId, collection]
+  )
+
+  useEffect(() => {
+    if (!selected) {
+      setError(fieldNamePrefix + 'collection', {
+        type: 'required',
+        message: 'no NFT selected',
+      })
+    } else {
+      clearErrors(fieldNamePrefix + 'collection')
+    }
+  }, [selected])
+
+  const getIdForNft = (nft: NftCardInfo) =>
+    `${nft.collection.address}:${nft.tokenId}`
+
+  const [showModal, setShowModal] = useState<boolean>(isCreating)
+
+  return (
+    <ActionCard
+      Icon={ImageEmoji}
+      onRemove={onRemove}
+      title={t('title.transferNft')}
+    >
+      <div className="flex flex-row flex-wrap items-center gap-2">
```

Comment:
What do you think about `justify-center` here to center horizontally on large screens? The left alignment looks weird. Also maybe the `NftCard` should be even smaller; not sure if it needs to be this size.

Reply:
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/30676292/205189839-158edd05-4eaf-4517-a2c0-1944e404e249.png">

i've tried a fair bit to make this look less bad, but i'm drawing a blank without making a pretty heavily modified NFT card which is more friendly to wide aspect ratios. maybe we can do that in followup work?